ros2topic/ros2topic/api/__init__.py

```diff
@@ -114,6 +115,19 @@ def set_msg_fields(msg, values):
         except ValueError as e:
             raise SetFieldError(field_name, e)
         try:
+            f_type = msg.get_slot_types()[field_name]
+            # Check if field is an array of ROS message types
+            if isinstance(field_type(), list):
+                if isinstance(f_type.basetype, NamespacedType):
+                    # TODO(mikaelarguedas) Consider putting this import logic in a rosidl function
```

Comment:
Maybe the new package from ros2/rosidl_python#34 would be a good place for this too.